### PR TITLE
Add image build steps in makefiles (with branch)

### PR DESCRIPTION
**Pull request by alknopfler**
Signed-off-by: Alberto Morgante Medina <alknopfler@users.noreply.github.com>

# Description

Add the possibility to build images from latest and from branch instead to fetch and clone from repo

Fixes [#96](https://github.com/rh-ecosystem-edge/ztp-pipeline-relocatable/issues/96)

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Test Makefile in local to make it before pushing to quay organization
- [ ] Test B

**Test Configuration**:

- Versions: linux
- Hardware: x86_platform

## Checklist

- [X] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [X] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Review by iranzo — APPROVED** [review]
nothing strange, lgtm to wait until CI pass to merge

**Comment by jparrill** [comment]
Pipeline passed, merging

**Pull request merged**
